backend/src/graphql/resolver/BalanceResolver.ts

```diff
@@ -5,36 +5,74 @@ import { Resolver, Query, Ctx, Authorized } from 'type-graphql'
 import { Balance } from '@model/Balance'
 import { calculateDecay } from '@/util/decay'
 import { RIGHTS } from '@/auth/RIGHTS'
-import { Transaction } from '@entity/Transaction'
+import { Transaction as dbTransaction } from '@entity/Transaction'
 import Decimal from 'decimal.js-light'
+import { GdtResolver } from './GdtResolver'
+import { TransactionLink as dbTransactionLink } from '@entity/TransactionLink'
+import { MoreThan, getCustomRepository } from '@dbTools/typeorm'
+import { TransactionLinkRepository } from '@repository/TransactionLink'
 
 @Resolver()
 export class BalanceResolver {
   @Authorized([RIGHTS.BALANCE])
   @Query(() => Balance)
   async balance(@Ctx() context: any): Promise<Balance> {
-    // load user and balance
     const { user } = context
     const now = new Date()
 
-    const lastTransaction = await Transaction.findOne(
-      { userId: user.id },
-      { order: { balanceDate: 'DESC' } },
-    )
+    const gdtResolver = new GdtResolver()
+    const balanceGDT = await gdtResolver.gdtBalance(context)
+
+    const lastTransaction = context.lastTransaction
+      ? context.lastTransaction
+      : await dbTransaction.findOne({ userId: user.id }, { order: { balanceDate: 'DESC' } })
 
     // No balance found
     if (!lastTransaction) {
       return new Balance({
         balance: new Decimal(0),
         decay: new Decimal(0),
-        decay_date: now.toString(),
+        lastBookedBalance: new Decimal(0),
+        balanceGDT,
+        count: 0,
+        linkCount: 0,
       })
     }
 
+    const count =
+      context.count || context.count === 0
```

Comment:
what is `context.count`? If we save stuff in the context at least have the courtesy to give it a name able to identify what it is (e.g. `transactionCount`)

Reply:
OK